src/pynxtools_raman/reader.py

```diff
@@ -1,300 +1,243 @@
+
+
+logger = logging.getLogger("pynxtools")
+
+CONVERT_DICT: Dict[str, str] = {}
+
+REPLACE_NESTED: Dict[str, str] = {}
+
+
+class RamanReader(MultiFormatReader):
+    """MyDataReader implementation for the DataConverter to convert mydata to NeXus."""
+
+    supported_nxdls = ["NXraman"]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.raman_data_dicts: List[Dict[str, Any]] = []
+        self.raman_data: Dict[str, Any] = {}
+        self.eln_data: Dict[str, Any] = {}
+        self.config_file: Path
+
+        self.missing_meta_data = None
+
+        self.extensions = {
+            ".yml": self.handle_eln_file,
+            ".yaml": self.handle_eln_file,
+            ".txt": self.handle_txt_file,
+            ".json": self.set_config_file,
+            ".rod": self.handle_rod_file,
+        }
+
+        # only required if multiple file types are present
+        # for ext in RamanReader.__prmt_file_ext__:
+        #    self.extensions[ext] = self.handle_data_file
+
+    def set_config_file(self, file_path: Path) -> Dict[str, Any]:
+        if self.config_file is not None:
+            logger.info(
+                f"Config file already set. Replaced by the new file {file_path}."
+            )
+        self.config_file = file_path
+        return {}
+
+    def handle_eln_file(self, file_path: str) -> Dict[str, Any]:
+        self.eln_data = parse_yml(
+            file_path,
+            convert_dict=CONVERT_DICT,
+            parent_key="/ENTRY[entry]",
+        )
+
+        return {}
+
+    def get_attr(self, key: str, path: str) -> Any:
+        """
+        Get the metadata that was stored in the main file.
+        """
+        return self.get_metadata(self.raman_data, path, self.callbacks.entry_name)
+
+    def read(
+        self,
+        template: dict = None,
+        file_paths: Tuple[str] = None,
+        objects: Tuple[Any] = None,
+        **kwargs,
+    ) -> dict:
+        template = super().read(template, file_paths, objects, suppress_warning=True)
+        # set default data
+
+        if self.missing_meta_data:
+            for key in self.missing_meta_data:
+                template[
+                    f"/ENTRY[{self.callbacks.entry_name}]/COLLECTION[unused_rod_keys]/{key}"
+                ] = f"{self.missing_meta_data[key]}"
+
+        template["/@default"] = "entry"
+
+        return template
+
+    def handle_rod_file(self, filepath) -> Dict[str, Any]:
+        # specify default config file for rod files
+        reader_dir = Path(__file__).parent
+        self.config_file = reader_dir.joinpath("config", "config_file_rod.json")  # pylint: disable=invalid-type-comment
+
+        rod = RodParser()
+        # read the rod file
+        rod.get_cif_file_content(filepath)
+        # get the key and value pairs from the rod file
+        self.raman_data = rod.extract_keys_and_values_from_cif()
+
+        self.missing_meta_data = copy.deepcopy(self.raman_data)

```

Comment:
I think for now it is fine, but you could also just make a copy of the keys and just reuse self.raman_data later.